RELEASING.md

```diff
@@ -0,0 +1,169 @@
+# Release Procedures
+
+*These notes are meant for a maintainer to create official releases.*
+
+In preparing a release, create a branch to hold pre-release commits. We ideally want all release mechanics (for all languages) to be in one commit, which will then be tagged. (This will change if/when we stop synchronizing releases across all language bindings.)
+
+## Core C library
+
+Some minor bookkeeping updates are needed when releasing a new version of the core library.
+
+The version number must be updated in
+
+* `include/ceed/ceed.h`
+* `ceed.pc.template`
+* `Doxyfile`
+
+Additionally, the release notes in `doc/sphinx/source/releasenotes.rst` should be updated. Use `git log --first-parent v0.7..` to get a sense of the pull requests that have been merged and thus might warrant emphasizing in the release notes. While doing this, gather a couple sentences for key features to highlight on [GitHub releases](https://github.com/CEED/libCEED/releases). The "Current Main" heading needs to be named for the release.
```

Comment:
Should we add any reference here in updating the User Manual version? That can easily fall out of date (as it is now)

Reply:
Updated now, and here's a PR for the users manual https://github.com/CEED/libCEED/pull/777

You can find the version history here.
https://zenodo.org/record/4895340

Reply:
Thanks Jed!